node/tessel.js

```diff
@@ -120,7 +121,19 @@ function Port(name, socketPath, board) {
     this.pin.G2 = this.pin.g2 = this.pin[6];
     this.pin.G3 = this.pin.g3 = this.pin[7];
     this.digital = [ this.pin[5], this.pin[6], this.pin[7] ];
+
     this.analog = [];
+    if (name == 'A') {
+        // pin 4 and pin 7 are adcs
+        this.analog.push(new AnalogPin(4, this));
+        this.analog.push(new AnalogPin(7, this));
+    } else if (name == 'B') {
+        // all pins are adcs
+        for (var i = 0; i < 8; i++) {
+            this.analog.push(new AnalogPin(i, this));
+        }
+    }
+
```

Comment:
Add the analog functionality to the `Pin` objects instead of creating a new class. I would like to deprecate the `analog` and `digital` arrays because they're a bad API. You never care about the "2nd analog pin" independent of its position; you care that your analog sensor is wired to pin 7 of the module port, which supports analog.